pkg/chunk/purger/tombstones.go

```diff
@@ -178,9 +243,53 @@ func (tl *TombstonesLoader) loadPendingTombstones(userID string) error {
 	defer tl.tombstonesMtx.Unlock()
 	tl.tombstones[userID] = &tombstoneSet
 
+	tl.metrics.deleteRequestsLoadFailures.Set(0)
 	return nil
 }
 
+// GetStoreCacheGenNumber returns store cache gen number for a user
+func (tl *TombstonesLoader) GetStoreCacheGenNumber(userID string) string {
```

Comment:
Can we pass a context here and propagate it, to ensure if the delete store is accessed we have context for the trace?